### PR TITLE
Enable X-Ray centralized sampling

### PR DESCRIPTION
**Pull request by vastin**
*Description of changes:*

Enable X-Ray centralized sampling according to https://aws-otel.github.io/docs/getting-started/remote-sampling

The code changes passed unit test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/instrumentation/podmutator.go [hunk 1]
```diff
@@ -21,20 +21,22 @@ import (
...
+	defaultExporterEndpoint                       = "http://amazon-cloudwatch-agent.amazon-cloudwatch:4317"
+	defaultJavaImage                              = "160148376629.dkr.ecr.us-west-2.amazonaws.com/aws-apm-preview:latest"
+	defaultAPIVersion                             = "cloudwatch.aws.amazon.com/v1alpha1"
+	defaultInstrumenation                         = "java-instrumentation"
+	defaultNamespace                              = "default"
+	defaultKind                                   = "Instrumentation"
+	otelSampleEnabledKey                          = "OTEL_SMP_ENABLED"
+	otelSampleEnabledDefaultValue                 = "true"
+	otelTracesSamplerArgKey                       = "OTEL_TRACES_SAMPLER_ARG"
+	otelTracesSamplerArgDefaultValue              = "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000"
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:33]: should be `amazon-cloudwatch-agent.amazon-cloudwatch`. the eks-addon defaults to this value, but in the future we will want this to be taken in as an arg to the operator as it's configurable...

**Review by lisguo — CHANGES_REQUESTED** [review]

File: pkg/instrumentation/podmutator.go [hunk 2]
```diff
@@ -21,20 +21,22 @@ import (
...
+	otelTracesSamplerKey                          = "OTEL_TRACES_SAMPLER"
+	otelTracesSamplerDefaultValue                 = "xray"
+	otelExporterTracesEndpointKey                 = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"
+	otelExporterTracesEndpointDefaultValue        = "http://amazon-cloudwatch-agent.amazon-cloudwatch:4317"
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:37]: the port should use `:4315`. Actually the above ports also.

**Review by lisguo — CHANGES_REQUESTED** [review]

File: pkg/instrumentation/podmutator.go (hunk 1 above)
Comment (vastin) [reply pkg/instrumentation/podmutator.go]: fixed

File: pkg/instrumentation/podmutator.go (hunk 2 above)
Comment (vastin) [reply pkg/instrumentation/podmutator.go]: fixed

**Comment by vastin** [comment]
Removed code change for span attribute limitation because it needs further investigation.

**Review by lisguo — APPROVED** [review]

**Review by dchappa — APPROVED** [review]

**Pull request merged**